bin/e2hs-writer/src/reader.rs

```diff
@@ -241,15 +244,28 @@ impl EpochReader {
     }
 
     pub fn iter_blocks(mut self) -> impl Iterator<Item = anyhow::Result<AllBlockData>> {
-        (self.starting_block..self.ending_block).map(move |current_block| match current_block {
-            0..MERGE_BLOCK_NUMBER => self.get_pre_merge_block_data(current_block),
-            MERGE_BLOCK_NUMBER..SHANGHAI_BLOCK_NUMBER => {
+        (self.starting_block..self.ending_block).map(move |current_block| {
+            if current_block
+                < EthereumHardfork::Paris
+                    .activation_block(network_spec().network().into())
+                    .expect("Paris is available")
```

Comment:
```suggestion
                    .expect("Paris should be available")
```

nit: If I would see the error/panic message "Paris is available", I would assume that is what caused the error (not what should be).

Same below.